### DISCUSSION TITLE
Copilot not aware of notebook context in VS Code

### DISCUSSION THREAD

**Discussion by LukeGahan** [community/community#62569] [Copilot Conversations] [answered]
**Edit: Regression happened in v1.99.289, with v1.98.275 being the last version to function as expected.**

  
  

**Expected Behaviour:** Copilot should be aware of code in the previous cells when providing suggestions in current cell.

**Current Behaviour:** Copilot is returning generic recommendations without awareness of previous cell. See the simple example below:

![image](https://github.com/community/community/assets/3843420/143c0bf3-4738-4620-ab5d-c95dc84a3a98)

**Details:**

Copilot extension: v1.100.306
Jupyter extension: v2023.6.1101941928
VS Code Version: June 2023 (version 1.80)

**Comment by tingyuansen** [comment]
I also face the same problem. Any update?

**Reply by Rockyread** [reply]
GitHub staff are lazy specially Akash. He never do a thing

**Comment by Spixmacaw** [comment]
![image](https://user-images.githubusercontent.com/3843420/258071343-143c0bf3-4738-4620-ab5d-c95dc84a3a98.png)

Expected Behaviour: Copilot should be aware of code in the previous cells when providing suggestions in current cell.

Current Behaviour: Copilot is returning generic recommendations without awareness of previous cell. See the simple example below:

Details:

Copilot extension: v1.100.306
Jupyter extension: v2023.6.1101941928
VS Code Version: June 2023 (version 1.80)

**Comment by NickTruesdale** [comment]
I'm seeing the same thing, on v1.102.333 (a release version) and v1.102.334 (a pre-release candidate). I feel like it was working within the past week - I wonder if we can pin down the release that caused the regression?

Edit: trying old versions of the Copilot VSCode extension doesn't fix this. It seems to be either an underlying problem with Copilot, or something introduced by a newer VSCode or Jupyter extension version?

**Reply by LukeGahan** [reply]
Yes it seems like this has been an issue for longer, to be honest. I can't remember exactly when the regression happened but it feels like 3 weeks ago or so...

edit: Seems we're not alone, other reports of the same issue [here on reddit](https://www.reddit.com/r/learnprogramming/comments/15ano7q/github_copilot_and_jupyter_notebook_suddenly/)

**Reply by Zhen3r** [reply]
Same, I downgraded copilot to v1.88.132, and the problem disappears. (with jupyter version v2023.6.1101941928)

**Reply by LukeGahan** [reply]
@Zhen3r Yes, rollback works for me too, I will leave this discussion open until the release version is working as expected. As per @bala-srm's comment below v1.98.275 also works, while v1.99.289 does not, so that's when the regression happened. If someone on the github team could fix it

**Reply by hsenag** [reply, marked as answer]
Sorry about this, and thanks for pinning down the affected versions. We believe it's fixed in 1.105, which has just been released. Please let us know if that's not the case!

**Reply by LukeGahan** [reply]
Yes it does seems to be fixed with that release, thank you!

**Reply by ivanpersan** [reply]
Copilot v1.105.361. Context still not working on notebooks for me.

**Reply by LukeGahan** [reply]
@ivanpersan Still working for me using the example above in 1.105

![image](https://github.com/community/community/assets/3843420/2bba2711-49c4-464c-abb5-663548fa9032)

**Comment by bala-srm** [comment]
Thanks for the update. That worked. In fact Copilot v1.98.275 works too with disappearance of the problem (with Jupyter version v2023.8.1002301100. From: Zhener ***@***.***>Date: Wednesday, 16 August 2023 at 12:37 PMTo: community/community ***@***.***>Cc: Balaji Sundararaman ***@***.***>, Manual ***@***.***>Subject: Re: [community/community] Copilot not aware of notebook context in VS Code (Discussion #62569)Same, I downgraded copilot to v1.88.132, and the problem disappears. (with jupyter version v2023.6.1101941928)—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

**Comment by crackwitz** [comment]
Still broken. Is this being worked on?

**Comment by GBdeMiranda** [comment]
I reopened my account this week after closing it for a month, and I wondered why the extension became so stupid.
It is disappointing to know that these issues were already reported three weeks ago, and still no bug fixes for a paid service.

**Comment by Rockyread** [comment]
GitHub staff are lazy specially Akash. He never do a thing

**Comment by bala-srm** [comment]
working for me as well with 1.105
